applications/GeoMechanicsApplication/tests/cpp_tests/test_newton_raphson_strategy_linear_elastic.cpp

```diff
@@ -0,0 +1,333 @@
+
+    auto rhs = Vector(2);
+    rhs(0)   = 0.0;
+    rhs(1)   = 10.0;
+
+    auto& geo_custom_element = dynamic_cast<GeoMockElement&>(model_part.GetElement(0));
+
+    geo_custom_element.SetLeftHandSide(stiffness_matrix);
+    geo_custom_element.SetMassMatrix(mass_matrix);
+    geo_custom_element.SetDampingMatrix(damping_matrix);
+
+    auto& geo_custom_condition = dynamic_cast<GeoMockCondition&>(model_part.GetCondition(0));
+
+    geo_custom_condition.SetRightHandSide(rhs);
+
+    // create strategy
+    auto r_solver = NewtonRaphsonStrategyLinearElasticDynamicTester::CreateValidStrategy(
+        model_part, RelativeConvergenceTolerance, AbsoluteConvergenceTolerance,
+        CalculateInitialAcceleration, UseDirectSolver);
+    // initialize solver
+    r_solver.Initialize();
+
+    std::vector<double> calculated_displacement_x;
+    std::vector<double> calculated_displacement_y;
+
+    std::size_t n_steps = 12;
+    // run test solution loop
+    for (std::size_t i = 0; i < n_steps; i++) {
+        double new_time = r_current_process_info[TIME] + r_current_process_info[DELTA_TIME];
+
+        r_current_process_info[STEP] += 1;
+        model_part.CloneTimeStep(new_time);
+        r_current_process_info[TIME] = new_time;
+
+        r_solver.InitializeSolutionStep();
+        r_solver.Predict();
+        r_solver.SolveSolutionStep();
+
+        r_solver.SetStiffnessMatrixIsBuilt(true);
+        r_solver.FinalizeSolutionStep();
+
+        // store calculated results
+        auto p_node = model_part.pGetNode(0);
+        calculated_displacement_x.push_back(p_node->FastGetSolutionStepValue(DISPLACEMENT_X, 0));
+        calculated_displacement_y.push_back(p_node->FastGetSolutionStepValue(DISPLACEMENT_Y, 0));
+    }
+
+    // check if the building functions are called the right amount, as this is largely why this solver is efficient for linear elastic systems
+    KRATOS_EXPECT_EQ(geo_custom_element.GetCountCalculateLeftHandSideCalled(), 1);
+    KRATOS_EXPECT_EQ(geo_custom_element.GetCountCalculateMassMatrixCalled(), 1);
+    KRATOS_EXPECT_EQ(geo_custom_element.GetCalculateDampingMatrixCalled(), 1);
+    KRATOS_EXPECT_EQ(geo_custom_element.GetCountCalculateRightHandSideCalled(), 0);
+
+    KRATOS_EXPECT_EQ(geo_custom_condition.GetCountCalculateLeftHandSideCalled(), 1);
+    KRATOS_EXPECT_EQ(geo_custom_condition.GetCountCalculateMassMatrixCalled(), 1);
+    KRATOS_EXPECT_EQ(geo_custom_condition.GetCalculateDampingMatrixCalled(), 1);
+
+    // rhs for conditions is called each solution step and during initialisation
+    if (UseIterations) {
+        // Two iterations are performed per solution step, thus rhs for conditions is called twice each solution step step and during initialisation
+        KRATOS_EXPECT_EQ(geo_custom_condition.GetCountCalculateRightHandSideCalled(), n_steps * 2 + 1);
+    } else {
+        KRATOS_EXPECT_EQ(geo_custom_condition.GetCountCalculateRightHandSideCalled(), n_steps + 1);
+    }
```

Comment:
Thanks for adding these checks, it also functions as documentation (it's really clear what is called how many times from this test)